packages/venia-concept/src/components/Navigation/categoryTree.js

```diff
@@ -21,45 +24,81 @@ class Tree extends Component {
+                    if (error) return <div>Data Fetch Error</div>;
+                    if (loading) return null;
 
-            const element = isLeaf ? (
-                <Leaf {...elementProps} onNavigate={onNavigate} />
-            ) : (
-                <Branch {...elementProps} onDive={updateRootNodeId} />
-            );
+                    const branches = [];
 
-            elements[position - 1] = <li key={id}>{element}</li>;
+                    const children = data.category.children.sort((a, b) => {
+                        if (a.position > b.position) return 1;
+                        else if (a.position == b.position && a.id > b.id)
+                            return 1;
+                        else return -1;
+                    });
 
-            return elements;
-        }, []);
+                    const leaves = children.map(node => {
+                        const { children_count } = node;
+                        const isLeaf = children_count == 0;
+                        const elementProps = {
+                            nodeId: node.id,
+                            name: node.name,
+                            urlPath: node.url_path,
+                            path: node.path
+                        };
 
-        if (nodes[rootNodeId].urlPath) {
-            leaves.push(
-                <li key={rootNodeId}>
-                    <Leaf
-                        nodeId={rootNodeId}
-                        nodes={nodes}
-                        onNavigate={onNavigate}
-                    >
-                        {({ node }) => `All ${node.name}`}
-                    </Leaf>
-                </li>
-            );
-        }
+                        if (!isLeaf) {
+                            branches.push(
+                                <CategoryTree
+                                    key={node.id}
+                                    rootNodeId={node.id}
+                                    updateRootNodeId={updateRootNodeId}
+                                    onNavigate={onNavigate}
+                                    currentId={currentId}
+                                />
+                            );
+                        }
 
-        return leaves;
+                        const element = isLeaf ? (
+                            <Leaf {...elementProps} onNavigate={onNavigate} />
+                        ) : (
+                            <Branch
+                                {...elementProps}
+                                onDive={updateRootNodeId}
+                            />
+                        );
+
+                        return <li key={node.id}>{element}</li>;
+                    });
+
+                    const branchClass =
+                        currentId == rootNodeId
+                            ? classes.branch
+                            : classes.inactive;
+                    return (
+                        <div>
```

Comment:
I feel like this div should have a class, but if it's totally styleable without, then nevermind.

Reply:
It doesn't serve any purpose, so it shouldn't even be a `div`. Changing it to a `Fragment`.